File: Abstraction/Parametric/abs.cs
using System;

class Program {
    public static T Abs<T>(T arg) where T : IComparable {
    	return arg.CompareTo(default(T)) > 0 ? arg : (dynamic)(-1) * arg;
	}  

    static void Main() {
        Console.WriteLine(Abs(4));
        Console.WriteLine(Abs(-3.5));
    }
}

File: Abstraction/Parametric/abs.rs
fn abs<T>(arg: T) -> T where T: PartialOrd + std::ops::Neg<Output = T> + Default {
    if arg > T::default() { arg } else { -arg }
}

pub fn main() {
    println!("{}", abs(4));
    println!("{}", abs(-3.6));
}
